backend/src/tools/web_pages.py

```diff
@@ -1,7 +1,35 @@
 import langchain_core.tools as lcct
+import bs4
+import requests as req
+import re
 
+# Should this perhaps be a separate, deterministic skill that we can plug into 
+#   any tool call that needs to clean text?
+def get_embedded_text(input: bs4.BeautifulSoup, strip:bool = False, separator: str = "") -> str:
+    pass
 
 @lcct.tool
-def parse_webpage(website: str) -> str:
-    """Mock: Return fake parsed content from website."""
-    return "Parsed content from the website (mocked)."
+def parse_webpage(website: str, embedded: bool = True) -> str:
+    """Parse a webpage and return its textual content."""
+
+    response = req.get(website)
+    response.raise_for_status()  # Raise an error for bad responses
+    content = bs4.BeautifulSoup(response.content, "html.parser")
+    strings = []
+    for descendant in content.descendants:
+        desc_type = None
```

Comment:
How do you use `desc_type`? it is defined in two places, but never used. I'm surprised that a linter didn't capture it for you.